### PR TITLE
feat: Update `artifacthub-pkg.yml` automatically from now

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Relates to https://github.com/kubewarden/kubewarden-controller/issues/325.

## Test

See CI run on this RP, and when the 0.1.23  tag is created.

To test this pull request, you can run the following commands:

### Potential improvement

- [ ] tag as 0.1.23 in main once it's merged.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
 
   test:
     name: run tests and linters
-    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v1
+    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v2
```
Comment (jvanz) [inline .github/workflows/release.yml:14]:
```suggestion
    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v2.0.2
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -23,6 +23,6 @@ jobs:
       # Required by cosign keyless signing
       id-token: write
 
-    uses: kubewarden/github-actions/.github/workflows/reusable-release-policy-go.yml@v1
+    uses: kubewarden/github-actions/.github/workflows/reusable-release-policy-go.yml@v2
```
Comment (jvanz) [inline .github/workflows/release.yml:26]:
```suggestion
    uses: kubewarden/github-actions/.github/workflows/reusable-release-policy-go.yml@v2.0.2
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -3,4 +3,4 @@ name: Continuous integration
 jobs:
   test:
     name: run tests and linters
-    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v1
+    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v2
```
Comment (jvanz) [inline .github/workflows/test.yml:6]:
```suggestion
    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v2.0.2
```

File: Makefile [hunk 4]
```diff
@@ -1,11 +1,23 @@
...
+		tinygo build -o policy.wasm -target=wasi -no-debug .
 
-annotated-policy.wasm: policy.wasm metadata.yml
-	kwctl annotate -m metadata.yml -o annotated-policy.wasm policy.wasm
+artifacthub-pkg.yml: metadata.yml go.mod
+	kwctl scaffold artifacthub \
+	    --metadata-path metadata.yml --version $(VERSION) \
+		--questions-path questions-ui.yml > artifacthub-pkg.yml.tmp \
+	&& mv artifacthub-pkg.yml.tmp artifacthub-pkg.yml \
+	|| rm -f artifacthub-pkg.yml.tmp
```
Comment (jvanz) [inline Makefile:17]:
```suggestion
artifacthub-pkg.yml: metadata.yml go.mod
        $(warning If you are updating the artifacthub-pkg.yml file for a release, \
                remember to set the VERSION variable with the proper value. \
                To use the latest tag, use the following command:  \
                make VERSION=$$(git describe --tags --abbrev=0 | cut -c2-) annotated-policy.wasm)
        kwctl scaffold artifacthub \
            --metadata-path metadata.yml --version $(VERSION) \
                --questions-path questions-ui.yml --output artifacthub-pkg.yml 
```

**Review by jvanz — CHANGES_REQUESTED** [review]
During the last planning we proposed some changes in the PRs updating the artifacthub files. I've opened a bunch of comments with these changes.

File: Makefile [hunk 5]
```diff
@@ -1,11 +1,23 @@
 SOURCE_FILES := $(shell find . -type f -name '*.go')
+VERSION := $(shell git describe --tags --abbrev=0 | cut -c2-)
```
Comment (jvanz) [inline Makefile:2]:
```suggestion
VERSION := $(shell git describe | cut -c2-)
```

**Comment by viccuad** [comment]
With the new proposed change to `VERSION`, the maximize the changes done to artifacthub-pkg.yml for those fields that consume version (oci urls, etc), as it now changes every commit. Hence we fail on CI here: https://github.com/kubewarden/ingress-policy/actions/runs/4467864309/jobs/7847991664#step:5:101

I will add that as another ignored line if we are not on the CI release job [here](https://github.com/kubewarden/github-actions/tree/ef34991cdbe29265c7ea494019c287cee3991334/check-artifacthub/action.yaml), and make a new release of github-actions.

File: .github/workflows/release.yml (hunk 1 above)
Comment (viccuad) [reply .github/workflows/release.yml]: Done in https://github.com/kubewarden/ingress-policy/pull/26/commits/5b9fb15a59a794fe1ed4914a4fe7e29f9f655ede!

**Pull request merged**